addons/autocomplete_googleplaces/static/src/js/addressForm.js

```diff
@@ -0,0 +1,84 @@
+/** @odoo-module */
+
+import publicWidget from 'web.public.widget';
+import { DropPrevious } from 'web.concurrency';
+import { debounce } from "@web/core/utils/timing";
+
+
+publicWidget.registry.AddressForm = publicWidget.Widget.extend({
+    selector: '.oe_cart .checkout_autoformat',
+    events: {
+        'input input[name="street"]': '_onChangeStreet',
+        'click .js_autocomplete_result': '_onClickAutocompleteResult'
+    },
+
+    init: function() {
+        this.streetAndNumberInput = document.querySelector('input[name="street"]');
+        this.cityInput = document.querySelector('input[name="city"]');
+        this.zipInput = document.querySelector('input[name="zip"]');
+        this.countrySelect = document.querySelector('select[name="country_id"]');
+        this.stateSelect = document.querySelector('select[name="state_id"]');
+        this.dp = new DropPrevious();
+
+        this._onChangeStreet = debounce(this._onChangeStreet, 200);
+        this._super.apply(this, arguments);
```

Comment:
I think we should check from the server whether it's useful to do the query, I am thinking specifically if you have the module but no api key. Or also in multi-website settings if for some reason you want it for one website but not another.